cylc/flow/task_pool.py

```diff
@@ -2066,34 +2073,47 @@ def _get_flow_nums(
             self,
             flow: List[str],
             meta: Optional[str] = None,
-    ) -> Optional[Set[int]]:
-        """Get correct flow numbers given user command options."""
-        if set(flow).intersection({FLOW_ALL, FLOW_NEW, FLOW_NONE}):
-            if len(flow) != 1:
-                LOG.warning(
-                    f'The "flow" values {FLOW_ALL}, {FLOW_NEW} & {FLOW_NONE}'
-                    ' cannot be used in combination with integer flow numbers.'
-                )
-                return None
-            if flow[0] == FLOW_ALL:
-                flow_nums = self._get_active_flow_nums()
-            elif flow[0] == FLOW_NEW:
-                flow_nums = {self.flow_mgr.get_flow_num(meta=meta)}
-            elif flow[0] == FLOW_NONE:
+            active: bool = False
+    ) -> Set[int]:
+        """Return flow numbers corresponding to user command options.
+
+        Arg should have been validated already during command validation.
+
+        Call this method separately for active (n=0) and future tasks.
+        - future tasks: assign the result to the new task
+        - active tasks: merge the result with existing flow numbers
+
+        Note if a single command results in two calls to this method (for
+        active and future tasks), translate --flow=new to an actual flow
+        number first, to avoid incrementing the flow counter twice.
```

Comment:
I don't like this. I've come up with what I think is a more elegant approach at https://github.com/hjoliver/cylc-flow/pull/56

Reply:
I agree it's not ideal. It was kind of a compromise on what I saw as a slightly bigger refactor. But I'll take a look at your way...